app/src/server/tasks/evaluateTestSetEntries.task.ts

```diff
@@ -0,0 +1,428 @@
+        errorMessage: null,
+      },
+    });
+
+    const entries = [];
+    const entryIds = [];
+
+    for (const result of [firstResult, secondResult]) {
+      try {
+        let entry;
+        if (result.datasetEvalOutputSource.modelId !== ORIGINAL_MODEL_ID) {
+          entry = await prisma.fineTuneTestingEntry.findUniqueOrThrow({
+            where: {
+              modelId_datasetEntryId: {
+                modelId: result.datasetEvalOutputSource.modelId,
+                datasetEntryId: result.datasetEvalDatasetEntry.datasetEntryId,
+              },
+            },
+            include: {
+              fineTune: true,
+            },
+          });
+          if (isComparisonModel(entry.modelId)) {
+            entryIds.push(getComparisonModelName(entry.modelId));
+          } else if (entry.fineTune) {
+            entryIds.push("openpipe:" + entry.fineTune.slug);
+          } else {
+            throw new Error("No fineTune or comparison model found for entry");
+          }
+        } else {
+          entry = result.datasetEvalDatasetEntry.datasetEntry;
+          entryIds.push("the original model");
+        }
+        entries.push(entry);
+      } catch (e) {
+        console.error("error getting entry for result", result, e);
+        await prisma.datasetEvalResult.updateMany({
+          where: {
+            id: {
+              in: [firstResult.id, secondResult.id],
+            },
+          },
+          data: {
+            status: "ERROR",
+            errorMessage: "Error retrieving relevant input for evaluation",
+          },
+        });
+        return;
+      }
+    }
+
+    const [firstEntry, secondEntry] = entries;
+    const [firstEntryId, secondEntryId] = entryIds;
+
+    if (!firstEntry?.output || !secondEntry?.output || !firstEntryId || !secondEntryId) {
+      await prisma.datasetEvalResult.updateMany({
+        where: {
+          id: {
+            in: [firstResult.id, secondResult.id],
+          },
+        },
+        data: {
+          status: "ERROR",
+          errorMessage: "Error preparing for evaluation",
+        },
+      });
+      return;
+    }
+
+    let explanation;
+    let judgement;
+
+    const instructions = firstResult.datasetEvalDatasetEntry.datasetEval.instructions;
+
+    try {
+      const input: ChatCompletionCreateParams = {
+        model: "gpt-4",
+        messages: [
+          {
+            role: "system",
+            content: `You are an intelligent and fair judge of chatbots. Evaluate the following two chatbot responses and choose which one is better in the following way: ${
+              instructions ?? ""
+            }`,
+          },
+          {
+            role: "user",
+            content: formatDatasetEntryInputInstructions(
+              firstResult.datasetEvalDatasetEntry.datasetEntry,
+            ),
+          },
+          {
+            role: "user",
+            content: `This is what Alice said:\n\n${JSON.stringify(firstEntry.output)}`,
```

Comment:
I'd lean towards just saying "This is response_1" and "This is response_2". Any reason to believe calling them "Alice" and "Bob" would work better? I'd be worried about that introducing more bias in the analysis, not less.

Reply:
@arcticfly any feedback on this?

Reply:
Personally, it would be easier for me to keep track of Alice's and Bob's answers rather than response_1 and response_2, which is why I leaned toward labeling the outputs that way. However, that advantage may not translate to GPT-4. I'd be totally fine with you experimenting with the wording here, and hopefully it won't require very much context loading.